src/clock/mumeclock.cpp

```diff
@@ -199,92 +149,100 @@ void MumeClock::parseMumeTime(const QString &mumeTime, const int64_t secsSinceEp
             + QString::number(newStartEpoch - m_mumeStartEpoch) + " seconds from previous)");
     } else {
-        log("Synchronized clock using 'time' output.");
+        log("Synchronized clock using 'time' output");
     }
     if (weekDay != capturedMoment.weekDay()) {
         qWarning() << "Calculated week day does not match MUME";
     }
     m_mumeStartEpoch = newStartEpoch;
 }
 
-void MumeClock::parseWeather(const QString &str)
+void MumeClock::slot_parseGmcpInput(const GmcpMessage &msg)
 {
-    const int64_t secsSinceEpoch = QDateTime::QDateTime::currentDateTimeUtc().toSecsSinceEpoch();
-    parseWeather(str, secsSinceEpoch);
-}
+    if (!(msg.isEventMoon() || msg.isEventDarkness() || msg.isEventSun())
+        || !msg.getJsonDocument().has_value() || !msg.getJsonDocument()->isObject())
+        return;
 
-void MumeClock::parseWeather(const QString &str, int64_t secsSinceEpoch)
-{
-    if (!m_stringTimeHash.contains(str)) {
+    const QJsonObject obj = msg.getJsonDocument()->object();
+    if (!obj.contains("what") || !obj.value("what").isString())
         return;
+
+    const auto type = obj["what"].toString();
+    if (type.isEmpty())
+        return;
+
+    MumeTimeEnum time = MumeTimeEnum::UNKNOWN;
+    if (msg.isEventSun()) {
+        switch (type.at(0).toLatin1()) {
+        case 'l': // light
+            time = MumeTimeEnum::DAY;
+            break;
+        case 'd': // dark
+            time = MumeTimeEnum::NIGHT;
+            break;
+        case 'r': // rise
+            time = MumeTimeEnum::DAWN;
+            break;
+        case 's': // set
+            time = MumeTimeEnum::DUSK;
+            break;
+        default:
+            assert(false);
+            break;
+        }
     }
+    const int64_t secsSinceEpoch = QDateTime::QDateTime::currentDateTimeUtc().toSecsSinceEpoch();
+    parseWeather(time, secsSinceEpoch);
+}
 
+void MumeClock::parseWeather(const MumeTimeEnum time, int64_t secsSinceEpoch)
+{
     // Update last sync timestamp
-    m_lastSyncEpoch = secsSinceEpoch;
+    setLastSyncEpoch(secsSinceEpoch);
 
-    const MumeTimeEnum time = m_stringTimeHash.value(str);
+    // Set minute to zero
     auto moment = MumeMoment::sinceMumeEpoch(secsSinceEpoch - m_mumeStartEpoch);
+    moment.minute = 0;
 
     // Predict current hour given the month
     const auto dawnDusk = getDawnDusk(moment.month);
     const int dawn = dawnDusk.dawnHour;
     const int dusk = dawnDusk.duskHour;
+
+    const char *type;
     switch (time) {
     case MumeTimeEnum::DAWN:
         moment.hour = dawn;
+        type = "sunrise";
         break;
     case MumeTimeEnum::DAY:
         moment.hour = dawn + 1;
+        type = "day";
         break;
     case MumeTimeEnum::DUSK:
         moment.hour = dusk;
+        type = "sunset";
         break;
     case MumeTimeEnum::NIGHT:
         moment.hour = dusk + 1;
+        type = "night";
         break;
     case MumeTimeEnum::UNKNOWN:
```

Comment:
Would it be better to just add WEATHER as a formal enum value?

Reply:
Weather was the previous string in the debug statement (will continue answer in next comment)

Reply:
I renamed this to "reason" to better identify it as a debugging feature